schema/object_test.go

```diff
@@ -94,6 +94,43 @@ func TestObjectUnserialization(t *testing.T) {
 	})
 }
 
+func TestObjectUnserialization_MissingFields(t *testing.T) {
+	dataMissing1 := map[string]any{
+		"field3": "Hello world!",
+	}
+	_, err := testStructSchema.Unserialize(dataMissing1)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "field is required")
+
+	dataMissing3 := map[string]any{
+		"Field1": 42,
+	}
+	_, err = testStructSchema.Unserialize(dataMissing3)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "field is required")
+}
```

Comment:
I think this is great -- but how hard would it be to confirm that it knows "field3" and then "Field1" are missing, rather than just that [a] field is required?